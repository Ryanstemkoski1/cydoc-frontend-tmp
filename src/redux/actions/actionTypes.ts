export enum CURRENT_NOTE_ACTION {
    LOAD_NOTE = 'CURRENT_NOTE_LOAD_NOTE',
    DELETE_NOTE = 'CURRENT_NOTE_DELETE_NOTE',
    UPDATE_NOTE_TITLE = 'CURRENT_NOTE_UPDATE_TITLE',
    UPDATE_NOTE_ID = 'CURRENT_NOTE_UPDATE_ID',
}

export enum ROS_ACTION {
    TOGGLE_OPTION = 'ROS_TOGGLE_OPTION',
}

export enum PHYSICAL_EXAM_ACTION {
    UPDATE_VITALS = 'PHYSICAL_EXAM_UPDATE_VITALS',
    TOGGLE_FINDING = 'PHYSICAL_EXAM_TOGGLE_FINDING',
    TOGGLE_LEFT_RIGHT_FINDING = 'PHYSICAL_EXAM_TOGGLE_LEFT_RIGHT_FINDING',
    TOGGLE_CHOOSE_BOOLEAN_VALUE = 'PHYSICAL_EXAM_TOGGLE_CHOOSE_BOOLEAN_VALUE',
    REMOVE_FINDING = 'PHYSICAL_EXAM_REMOVE_FINDING',
    UPDATE_COMMENTS = 'PHYSICAL_EXAM_UPDATE_COMMENTS',
}

export enum LUNGS_WIDGET_ACTION {
    TOGGLE_LUNGS_WIDGET_SECTION = 'LUNGS_WIDGET_TOGGLE_LUNGS_WIDGET_SECTION',
}

export enum ABDOMEN_WIDGET_ACTION {
    TOGGLE_ABDOMEN_WIDGET_SECTION = 'ABDOMEN_WIDGET_TOGGLE_ABDOMEN_WIDGET_SECTION',
}

export enum PULSES_WIDGET_ACTION {
    ADD_PULSES_WIDGET_ITEM = 'PULSES_WIDGET_ADD_PULSES_WIDGET_ITEM',
    UPDATE_LOCATION = 'PULSES_WIDGET_UPDATE_LOCATION',
    UPDATE_SIDE = 'PULSES_WIDGET_UPDATE_SIDE',
    UPDATE_INTENSITY = 'PULSES_WIDGET_UPDATE_INTENSITY',
    DELETE_PULSES_WIDGET_ITEM = 'PULSES_WIDGET_DELETE_PULSES_WIDGET_ITEM',
}

export enum REFLEXES_WIDGET_ACTION {
    ADD_REFLEXES_WIDGET_ITEM = 'REFLEXES_WIDGET_ADD_REFLEXES_WIDGET_ITEM',
    UPDATE_LOCATION = 'REFLEXES_WIDGET_UPDATE_LOCATION',
    UPDATE_SIDE = 'REFLEXES_WIDGET_UPDATE_SIDE',
    UPDATE_INTENSITY = 'REFLEXES_WIDGET_UPDATE_INTENSITY',
    DELETE_REFLEXES_WIDGET_ITEM = 'REFLEXES_WIDGET_DELETE_REFLEXES_WIDGET_ITEM',
}

export enum MURMURS_WIDGET_ACTION {
    ADD_MURMURS_WIDGET_ITEM = 'MURMURS_WIDGET_ADD_MURMURS_WIDGET_ITEM',
    UPDATE_PHASE = 'MURMURS_WIDGET_UPDATE_PHASE',
    TOGGLE_CRESCENDO_DECRESCENDO = 'MURMURS_WIDGET_TOGGLE_CRESCENDO_DECRESCENDO',
    UPDATE_BEST_HEARD_AT = 'MURMURS_WIDGET_UPDATE_BEST_HEARD_AT',
    UPDATE_INTENSITY = 'MURMURS_WIDGET_UPDATE_INTENSITY',
    UPDATE_PITCH = 'MURMURS_WIDGET_UPDATE_PITCH',
    TOGGLE_QUALITY = 'MURMURS_WIDGET_TOGGLE_QUALITY',
    TOGGLE_SPECIFIC_MURMUR_INFO = 'MURMURS_WIDGET_TOGGLE_SPECIFIC_MURMUR_INFO',
    UPDATE_SPECIFIC_MURMUR = 'MURMURS_WIDGET_UPDATE_SPECIFIC_MURMUR',
    TOGGLE_RADIATION_TO = 'MURMURS_WIDGET_TOGGLE_RADIATION_TO',
    TOGGLE_ADDITIONAL_FEATURES = 'MURMURS_WIDGET_TOGGLE_ADDITIONAL_FEATURES',
    DELETE_MURMURS_WIDGET_ITEM = 'MURMURS_WIDGET_DELETE_MURMURS_WIDGET_ITEM',
}
export enum MEDICAL_HISTORY_ACTION {
    TOGGLE_OPTION = 'MEDICAL_HISTORY_TOGGLE_OPTION',
    UPDATE_CONDITION_NAME = 'MEDICAL_HISTORY_UPDATE_CONDITION_NAME',
    UPDATE_START_YEAR = 'MEDICAL_HISTORY_UPDATE_START_YEAR',
    UPDATE_CONDITION_RESOLVED = 'MEDICAL_HISTORY_UPDATE_CONDITION_RESOLVED',
    UPDATE_END_YEAR = 'MEDICAL_HISTORY_UPDATE_END_YEAR',
    UPDATE_COMMENTS = 'MEDICAL_HISTORY_UPDATE_COMMENTS',
    ADD_PMH_POP_OPTIONS = 'MEDICAL_HISTORY_ADD_PMH_POP_OPTIONS',
    DELETE_CONDITION = 'MEDICAL_HISTORY_DELETE_CONDITION',
}

export enum SURGICAL_HISTORY_ACTION {
    UPDATE_PROCEDURE = 'SURGICAL_HISTORY_UPDATE_PROCEDURE',
    TOGGLE_OPTION = 'SURGICAL_HISTORY_TOGGLE_OPTION',
    UPDATE_YEAR = 'SURGICAL_HISTORY_UPDATE_YEAR',
    UPDATE_COMMENTS = 'SURGICAL_HISTORY_UPDATE_COMMENTS',
    DELETE_PROCEDURE = 'SURGICAL_HISTORY_DELETE_PROCEDURE',
    ADD_PSH_POP_OPTIONS = 'SURGICAL_HISTORY_ADD_PSH_POP_OPTIONS',
}

export enum MEDICATIONS_ACTION {
    UPDATE_DRUG_NAME = 'MEDICATIONS_UPDATE_DRUG_NAME',
    UPDATE_START_YEAR = 'MEDICATIONS_UPDATE_START_YEAR',
    UPDATE_CURRENTLY_TAKING = 'MEDICATIONS_UPDATE_CURRENTLY_TAKING',
    UPDATE_END_YEAR = 'MEDICATIONS_UPDATE_END_YEAR',
    UPDATE_SCHEDULE = 'MEDICATIONS_SCHEDULE',
    UPDATE_DOSE = 'MEDICATIONS_UPDATE_DOSE',
    UPDATE_REASON_FOR_TAKING = 'MEDICATIONS_UPDATE_REASON_FOR_TAKING',
    UPDATE_SIDE_EFFECTS = 'MEDICATIONS_UPDATE_SIDE_EFFECTS',
    UPDATE_COMMENTS = 'MEDICATIONS_UPDATE_COMMENTS',
    DELETE_MEDICATION = 'MEDICATIONS_DELETE_MEDICATION',
    ADD_MEDS_POP_OPTION = 'MEDICATIONS_ADD_MEDS_POP_OPTION',
}

export enum ALLERGIES_ACTION {
    UPDATE_INCITING_AGENT = 'ALLERGIES_UPDATE_INCITING_AGENT',
    UPDATE_REACTION = 'ALLERGIES_UPDATE_REACTION',
    UPDATE_COMMENTS = 'ALLERGIES_UPDATE_COMMENTS',
    UPDATE_ID = 'ALLERGIES_UPDATE_ID',
    ADD_ALLERGY = 'ALLERGIES_ADD_ALLERGY',
    DELETE_ALLERGY = 'ALLERGIES_DELETE_ALLERGY',
}

export enum SOCIAL_HISTORY_ACTION {
    UPDATE_ALCOHOL_USAGE = 'SOCIAL_HISTORY_UPDATE_A_USAGE',
    ADD_ALCOHOL_CONSUMPTION = 'SOCIAL_HISTORY_ADD_A_CONSUMPTION',
    UPDATE_ALCOHOL_CONSUMPTION_TYPE = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_TYPE',
    UPDATE_ALCOHOL_CONSUMPTION_SIZE = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_SIZE',
    UPDATE_ALCOHOL_CONSUMPTION_NUMBER_PER_WEEK = 'SOCIAL_HISTORY_UPDATE_A_CONSUMPTION_PER_WEEK',
    DELETE_ALCOHOL_CONSUMPTION = 'SOCIAL_HISTORY_DELETE_A_CONSUMPTION',
    UPDATE_ALCOHOL_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_A_QUIT_YEAR',
    UPDATE_ALCOHOL_COMMENTS = 'SOCIAL_HISTORY_UPDATE_A_COMMENTS',
    UPDATE_ALCOHOL_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_A_IIQ',
    UPDATE_ALCOHOL_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_A_TTQ',

    UPDATE_TOBACCO_USAGE = 'SOCIAL_HISTORY_UPDATE_T_USAGE',
    UPDATE_TOBACCO_PACKS_PER_DAY = 'SOCIAL_HISTORY_UPDATE_T_PACKS_PER_DAY',
    UPDATE_TOBACCO_NUMBER_OF_YEARS = 'SOCIAL_HISTORY_UPDATE_T_NUMBER_OF_YEARS',
    UPDATE_TOBACCO_PRODUCTS_USED = 'SOCIAL_HISTORY_UPDATE_T_PRODUCTS_USED',
    UPDATE_TOBACCO_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_T_QUIT_YEAR',
    UPDATE_TOBACCO_COMMENTS = 'SOCIAL_HISTORY_UPDATE_T_COMMENTS',
    UPDATE_TOBACCO_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_T_IIQ',
    UPDATE_TOBACCO_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_T_TTQ',

    UPDATE_RECREATIONAL_DRUG_USAGE = 'SOCIAL_HISTORY_UPDATE_RD_USAGE',
    ADD_RECREATIONAL_DRUG_USED = 'SOCIAL_HISTORY_ADD_RD_USED',
    UPDATE_RECREATIONAL_DRUG_USED_NAME = 'SOCIAL_HISTORY_UPDATE_RD_USED_NAME',
    UPDATE_RECREATIONAL_DRUG_USED_MODES_OF_DELIVERY = 'SOCIAL_HISTORY_UPDATE_RD_USED_MOD',
    UPDATE_RECREATIONAL_DRUG_USED_NUMBER_PER_WEEK = 'SOCIAL_HISTORY_UPDATE_RD_USED_PER_WEEK',
    DELETE_RECREATIONAL_DRUG_USED = 'SOCIAL_HISTORY_DELETE_RD_USED',
    UPDATE_RECREATIONAL_DRUG_QUIT_YEAR = 'SOCIAL_HISTORY_UPDATE_RD_QUIT_YEAR',
    UPDATE_RECREATIONAL_DRUG_COMMENTS = 'SOCIAL_HISTORY_UPDATE_RD_COMMENTS',
    UPDATE_RECREATIONAL_DRUG_INTERESTED_IN_QUITTING = 'SOCIAL_HISTORY_UPDATE_RD_IIQ',
    UPDATE_RECREATIONAL_DRUG_TRIED_TO_QUIT = 'SOCIAL_HISTORY_UPDATE_RD_TTQ',

    UPDATE_LIVING_SITUATION = 'SOCIAL_HISTORY_UPDATE_LIVING_SITUATION',
    UPDATE_EMPLOYMENT = 'SOCIAL_HISTORY_UPDATE_EMPLOYMENT',
    UPDATE_DIET = 'SOCIAL_HISTORY_UPDATE_DIET',
    UPDATE_EXERCISE = 'SOCIAL_HISTORY_UPDATE_EXERCISE',
}

export enum FAMILY_HISTORY_ACTION {
    TOGGLE_CONDITION_OPTION = 'FAMILY_HISTORY_TOGGLE_CONDITION_OPTION',
    ADD_FAMILY_MEMBER = 'FAMILY_HISTORY_ADD_FAMILY_MEMBER',
    DELETE_FAMILY_MEMBER = 'FAMILY_HISTORY_DELETE_FAMILY_MEMBER',
    UPDATE_MEMBER = 'FAMILY_HISTORY_UPDATE_MEMBER',
    TOGGLE_CAUSE_OF_DEATH_OPTION = 'FAMILY_HISTORY_TOGGLE_CAUSE_OF_DEATH_OPTION',
    TOGGLE_LIVING_OPTION = 'FAMILY_HISTORY_TOGGLE_LIVING_OPTION',
    UPDATE_COMMENTS = 'FAMILY_HISTORY_UPDATE_COMMENTS',
    UPDATE_CONDITION_NAME = 'FAMILY_HISTORY_UPDATE_CONDITION_NAME',
    ADD_FH_POP_OPTIONS = 'FAMILY_HISTORY_ADD_FH_POP_OPTIONS',
    DELETE_CONDITION = 'FAMILY_HISTORY_DELETE_CONDITION',
}

export enum PLAN_ACTION {
    ADD_CONDITION = 'PLAN_ADD_CONDITION',
    DELETE_CONDITION = 'PLAN_DELETE_CONDITION',
    UPDATE_CONDITION_NAME = 'PLAN_UPDATE_CONDITION_NAME',

    ADD_DIFFERENTIAL_DIAGNOSIS = 'PLAN_ADD_DIFFERENTIAL_DIAGNOSIS',
    UPDATE_DIFFERENTIAL_DIAGNOSIS_COMMENTS = 'PLAN_UPDATE_DIFFERENTIAL_DIAGNOSIS_COMMENTS',
    UPDATE_DIFFERENTIAL_DIAGNOSIS = 'PLAN_UPDATE_DIFFERENTIAL_DIAGNOSIS',
    DELETE_DIFFERENTIAL_DIAGNOSIS = 'PLAN_DELETE_DIFFERENTIAL_DIAGNOSIS',

    ADD_PRESCRIPTION = 'PLAN_ADD_PRESCRIPTION',
    UPDATE_PRESCRIPTION_COMMENTS = 'PLAN_UPDATE_PRESCRIPTION_COMMENTS',
    UPDATE_PRESCRIPTION_DOSE = 'PLAN_UPDATE_PRESCRIPTION_DOSE',
    UPDATE_PRESCRIPTION_TYPE = 'PLAN_UPDATE_PRESCRIPTION_TYPE',
    UPDATE_PRESCRIPTION_SIGNATURE = 'PLAN_UPDATE_PRESCRIPTION_SIGNATURE',
    DELETE_PRESCRIPTION = 'PLAN_DELETE_PRESCRIPTION',

    ADD_PROCEDURE_OR_SERVICE = 'PLAN_ADD_PROCEDURE_OR_SERVICE',
    UPDATE_PROCEDURE_OR_SERVICE = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE',
    UPDATE_PROCEDURE_OR_SERVICE_COMMENTS = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE_COMMENTS',
    UPDATE_PROCEDURE_OR_SERVICE_WHEN = 'PLAN_UPDATE_PROCEDURE_OR_SERVICE_WHEN',
    DELETE_PROCEDURE_OR_SERVICE = 'PLAN_DELETE_PROCEDURE_OR_SERVICE',

    ADD_REFERRAL = 'PLAN_ADD_REFERRAL',
    UPDATE_REFERRAL_DEPARTMENT = 'PLAN_UPDATE_REFERRAL_DEPARTMENT',
    UPDATE_REFERRAL_COMMENTS = 'PLAN_UPDATE_REFERRAL_COMMENTS',
    UPDATE_REFERRAL_WHEN = 'PLAN_UPDATE_REFERRAL_WHEN',
    DELETE_REFERRAL = 'PLAN_DELETE_REFERRAL',

    UPDATE_ADMIT_TO_HOSPITAL = 'PLAN_UPDATE_ADMIT_TO_HOSPITAL',
    UPDATE_EMERGENCY = 'PLAN_UPDATE_EMERGENCY',
    UPDATE_SICKNESS_LEVEL = 'PLAN_UPDATE_SICKNESS_LEVEL',
}

export enum HPI_ACTION {
    PROCESS_KNOWLEDGE_GRAPH = 'HPI_PROCESS_KNOWLEDGE_GRAPH',
    BODY_LOCATION_RESPONSE = 'BODY_LOCATION_RESPONSE',
    BODY_LOCATION_HANDLE_TOGGLE = 'HPI_BODY_LOCATION_HANDLE_TOGGLE',
    SINGLE_MULTIPLE_CHOICE_HANDLE_CLICK = 'SINGLE_MULTIPLE_CHOICE_HANDLE_CLICK',
    HANDLE_INPUT_CHANGE = 'HPI_HANDLE_INPUT_CHANGE',
    HANDLE_NUMERIC_INPUT_CHANGE = 'HANDLE_NUMERIC_INPUT_CHANGE',
    LIST_TEXT_HANDLE_CHANGE = 'HPI_LIST_TEXT_HANDLE_CHANGE',
    REMOVE_LIST_INPUT = 'HPI_LIST_TEXT_REMOVE_INPUT',
    ADD_LIST_INPUT = 'HPI_LIST_TEXT_ADD_INPUT',
    HANDLE_TIME_INPUT_CHANGE = 'HPI_HANDLE_TIME_INPUT_CHANGE',
    HANDLE_TIME_OPTION_CHANGE = 'HPI_HANDLE_TIME_OPTION_CHANGE',
    HANDLE_TIME_OF_DAY_CHANGE = 'HPI_HANDLE_TIME_OF_DAY_CHANGE',
    YES_NO_TOGGLE_OPTION = 'HPI_YES_NO_TOGGLE_OPTION',
    SCALE_HANDLE_VALUE = 'HPI_SCALE_HANDLE_VALUE',
    SCALE_HANDLE_CLEAR = 'HPI_SCALE_HANDLE_CLEAR',
    HANDLE_BLANK_QUESTION_CHANGE = 'HPI_HANDLE_BLANK_QUESTION_CHANGE',
    POP_RESPONSE = 'HPI_POP_RESPONSE',
    LAB_TEST_INPUT_CHANGE = 'HPI_LAB_TEST_INPUT_CHANGE',
    LAB_TEST_HANDLE_CLICK = 'HPI_LAB_TEST_HANDLE_CLICK',
    MEDS_POP_YES_NO_TOGGLE = 'HPI_MEDS_POP_YES_NO_TOGGLE',
}

export enum CHIEF_COMPLAINTS {
    SELECT_CHIEF_COMPLAINTS = 'SELECT_CHIEF_COMPLAINTS',
    SET_NOTES_CHIEF_COMPLAINTS = 'SET_NOTES_CHIEF_COMPLAINTS',
}

export enum PATIENT_INFORMATION_ACTION {
    UPDATE_INFORMATION = 'UPDATE_INFORMATION',
    UPDATE_PRONOUNS = 'UPDATE_PRONOUNS',
    UPDATE_NAME = 'UPDATE_NAME',
}

export enum HPI_HEADER_ACTION {
    SAVE_HPI_HEADER = 'SAVE_HPI_HEADER',
}
