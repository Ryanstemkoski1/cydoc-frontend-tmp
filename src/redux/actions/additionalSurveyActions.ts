import { UPDATE_ADDITIONAL_DETAILS } from './actionTypes';

export interface UpdateAdditionalSurveyAction {
    type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS;
    payload: {
        legalFirstName: string;
        legalLastName: string;
        socialSecurityNumber: string;
        dateOfBirth: string;
        showAdditionalSurvey: boolean;
    };
}

export interface GoBackToAdditionalSurvey {
    type: UPDATE_ADDITIONAL_DETAILS.GO_BACK;
    payload: {
        showAdditionalSurvey: boolean;
    };
}

export function updateAdditionalSurveyDetails(
    legalFirstName: string,
    legalLastName: string,
    socialSecurityNumber: string,
    dateOfBirth: string
): UpdateAdditionalSurveyAction {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.UPDATE_ADDITIONAL_DETAILS,
        payload: {
            legalFirstName,
            legalLastName,
            socialSecurityNumber,
            dateOfBirth,
            showAdditionalSurvey: false,
        },
    };
}

export function resetAdditionalSurveyPage(): GoBackToAdditionalSurvey {
    return {
        type: UPDATE_ADDITIONAL_DETAILS.GO_BACK,
        payload: {
            showAdditionalSurvey: true,
        },
    };
}
