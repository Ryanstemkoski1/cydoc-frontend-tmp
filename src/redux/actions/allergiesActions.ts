import { ALLERGIES_ACTION } from './actionTypes';

interface HasAllergiesAction {
    type: ALLERGIES_ACTION.HAS_ALLERGIES;
    payload: {
        hasAllergies: boolean;
    };
}

export function toggleHasAllergies(state: boolean): HasAllergiesAction {
    return {
        type: ALLERGIES_ACTION.HAS_ALLERGIES,
        payload: {
            hasAllergies: state,
        },
    };
}

interface UpdateIncitingAgentAction {
    type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT;
    payload: {
        index: string;
        newIncitingAgent: string;
    };
}

export function updateIncitingAgent(index: string, newIncitingAgent: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_INCITING_AGENT,
        payload: {
            index,
            newIncitingAgent,
        },
    };
}

interface UpdateReactionAction {
    type: ALLERGIES_ACTION.UPDATE_REACTION;
    payload: {
        index: string;
        newReaction: string;
    };
}

export function updateReaction(index: string, newReaction: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_REACTION,
        payload: {
            index,
            newReaction,
        },
    };
}

interface UpdateCommentsAction {
    type: ALLERGIES_ACTION.UPDATE_COMMENTS;
    payload: {
        index: string;
        newComments: string;
    };
}

export function updateComments(index: string, newComments: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_COMMENTS,
        payload: {
            index,
            newComments,
        },
    };
}

interface UpdateIdAction {
    type: ALLERGIES_ACTION.UPDATE_ID;
    payload: {
        index: string;
        id: string;
    };
}

export function updateId(index: string, id: string) {
    return {
        type: ALLERGIES_ACTION.UPDATE_ID,
        payload: {
            index,
            id,
        },
    };
}

interface AddAllergyAction {
    type: ALLERGIES_ACTION.ADD_ALLERGY;
}

export function addAllergy() {
    return {
        type: ALLERGIES_ACTION.ADD_ALLERGY,
    };
}

interface DeleteAllergyAction {
    type: ALLERGIES_ACTION.DELETE_ALLERGY;
    payload: {
        index: string;
    };
}

export function deleteAllergy(index: string) {
    return {
        type: ALLERGIES_ACTION.DELETE_ALLERGY,
        payload: {
            index,
        },
    };
}

export type AllergiesActionTypes =
    | HasAllergiesAction
    | UpdateIncitingAgentAction
    | UpdateReactionAction
    | UpdateCommentsAction
    | UpdateIdAction
    | AddAllergyAction
    | DeleteAllergyAction;
