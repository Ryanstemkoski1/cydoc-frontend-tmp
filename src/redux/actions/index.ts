import { ROSActionTypes } from './reviewOfSystemsActions';
import { MedicalHistoryActionTypes } from './medicalHistoryActions';
import { SurgicalHistoryActionTypes } from './surgicalHistoryActions';
import { MedicationsActionTypes } from './medicationsActions';
import { AllergiesActionTypes } from './allergiesActions';
import { SocialHistoryActionTypes } from './socialHistoryActions';
import { FamilyHistoryActionTypes } from './familyHistoryActions';
import { PhysicalExamActionTypes } from './physicalExamActions';
import { PlanActionTypes } from './planActions';
import { HpiActionTypes } from './hpiActions';
import { WidgetActionTypes } from './widgetActions';
import { chiefComplaintsActionTypes } from './chiefComplaintsActions';

export type AllActionTypes =
    | ROSActionTypes
    | PhysicalExamActionTypes
    | WidgetActionTypes
    | MedicalHistoryActionTypes
    | SurgicalHistoryActionTypes
    | MedicationsActionTypes
    | AllergiesActionTypes
    | SocialHistoryActionTypes
    | FamilyHistoryActionTypes
    | PlanActionTypes
    | HpiActionTypes
    | chiefComplaintsActionTypes;
