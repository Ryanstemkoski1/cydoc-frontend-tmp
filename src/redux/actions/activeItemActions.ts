import { UnknownAction } from 'redux';
import { ACTIVE_ITEM_ACTION } from './actionTypes';

export interface UpdateActiveItemAction extends UnknownAction {
    type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM;
    payload: {
        updatedItem: string;
    };
}

export function updateActiveItem(updatedItem: string): UpdateActiveItemAction {
    return {
        type: ACTIVE_ITEM_ACTION.ACTIVE_ITEM,
        payload: {
            updatedItem,
        },
    };
}

export type activeItemActionTypes = UpdateActiveItemAction;
