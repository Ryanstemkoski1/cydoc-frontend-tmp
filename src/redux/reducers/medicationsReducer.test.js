import {
    medicationsReducer,
    initialMedicationsState,
} from './medicationsReducer';
import { MEDICATIONS_ACTION } from 'redux/actions/actionTypes';
import { YesNoResponse } from 'constants/enums';

const initialMedications = {
    uuid1: {
        drugName: '',
        startYear: -1,
        isCurrentlyTaking: YesNoResponse.None,
        endYear: -1,
        schedule: '',
        dose: '',
        reasonForTaking: '',
        sideEffects: [],
        comments: '',
    },
};

describe('medications reducers', () => {
    it('returns the initial state', () => {
        expect(medicationsReducer(undefined, {})).toEqual(
            initialMedicationsState
        );
    });
    it('updates drug name', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_DRUG_NAME,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newDrugName: 'newDrugName',
                },
            })
        ).toMatchSnapshot();
    });
    it('updates start year', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_START_YEAR,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newStartYear: 2019,
                },
            })
        ).toMatchSnapshot();
    });
    it('updates currently taking', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_CURRENTLY_TAKING,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    optionSelected: YesNoResponse.No,
                },
            })
        ).toMatchSnapshot();
    });
    it('updates end year', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_END_YEAR,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newEndYear: 2020,
                },
            })
        ).toMatchSnapshot();
    });
    it('updates schedule', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_SCHEDULE,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newSchedule: 'newSchedule',
                },
            })
        ).toMatchSnapshot();
    });
    it('updates dose', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_DOSE,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newDose: 'newDose',
                },
            })
        ).toMatchSnapshot();
    });
    it('updates reason for taking', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_REASON_FOR_TAKING,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newReasonForTaking: 'newReasonForTaking',
                },
            })
        ).toMatchSnapshot();
    });
    it('updates side effects', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_SIDE_EFFECTS,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newSideEffects: ['newSideEffect'],
                },
            })
        ).toMatchSnapshot();
    });
    it('updates comments', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.UPDATE_COMMENTS,
                payload: {
                    index: Object.keys(initialMedications)[0],
                    newComments: 'newComments',
                },
            })
        ).toMatchSnapshot();
    });
    it('deletes medication entry', () => {
        expect(
            medicationsReducer(initialMedications, {
                type: MEDICATIONS_ACTION.DELETE_MEDICATION,
                payload: {
                    index: 'uuid1',
                },
            })
        ).toMatchSnapshot();
    });
    it('creates new entry with given condition id', () => {
        const nextState = medicationsReducer(initialMedications, {
            type: MEDICATIONS_ACTION.ADD_MEDS_POP_OPTION,
            payload: {
                medIndex: 'foo',
                medName: 'foo',
            },
        });
        expect(nextState).toMatchSnapshot();
        expect(nextState).toHaveProperty('foo');
        expect(nextState['foo'].drugName).toEqual('foo');
    });
});
