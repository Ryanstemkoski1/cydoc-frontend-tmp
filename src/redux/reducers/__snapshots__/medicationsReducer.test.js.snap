// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`medications reducers creates new entry with given condition id 1`] = `
Object {
  "foo": Object {
    "comments": "",
    "dose": "",
    "drugName": "foo",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers deletes medication entry 1`] = `Object {}`;

exports[`medications reducers updates comments 1`] = `
Object {
  "uuid1": Object {
    "comments": "newComments",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates currently taking 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "NO",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates dose 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "newDose",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates drug name 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "newDrugName",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates end year 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": 2020,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates reason for taking 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "newReasonForTaking",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates schedule 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "newSchedule",
    "sideEffects": Array [],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates side effects 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [
      "newSideEffect",
    ],
    "startYear": -1,
  },
}
`;

exports[`medications reducers updates start year 1`] = `
Object {
  "uuid1": Object {
    "comments": "",
    "dose": "",
    "drugName": "",
    "endYear": -1,
    "isCurrentlyTaking": "",
    "reasonForTaking": "",
    "schedule": "",
    "sideEffects": Array [],
    "startYear": 2019,
  },
}
`;
